packages/snaps-controllers/src/interface/InterfaceController.ts

```diff
@@ -0,0 +1,219 @@
+  state?: InterfaceControllerState;
+};
+
+/**
+ * Use this controller to manage snaps UI interfaces using RPC method hooks.
+ */
+export class InterfaceController extends BaseController<
+  typeof controllerName,
+  InterfaceControllerState,
+  InterfaceControllerMessenger
+> {
+  constructor({ messenger, state }: InterfaceControllerArgs) {
+    super({
+      messenger,
+      metadata: {
+        interfaces: { persist: false, anonymous: false },
+      },
+      name: controllerName,
+      state: { interfaces: {}, ...state },
+    });
+
+    this.#registerMessageHandlers();
+  }
+
+  /**
+   * Constructor helper for registering this controller's messaging system
+   * actions.
+   */
+  #registerMessageHandlers() {
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:createInterface`,
+      this.createInterface.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:getInterface`,
+      this.getInterface.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:updateInterface`,
+      this.updateInterface.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:deleteInterface`,
+      this.deleteInterface.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:updateInterfaceState`,
+      this.updateInterfaceState.bind(this),
+    );
+  }
+
+  /**
+   * Creates an interface in the controller state with the associated data.
+   *
+   * @param snapId - The snap id that created the interface.
+   * @param content - The interface content.
+   * @returns The newly interface id.
+   */
+  createInterface(snapId: string, content: Component) {
+    const id = nanoid();
+
+    const componentState = constructState({}, content);
+
+    this.update((draftState) => {
+      draftState.interfaces[id] = {
+        snapId,
+        content,
+        state: componentState,
+      };
+    });
+
+    return id;
+  }
+
+  /**
+   * Get the data of a given interface id.
+   *
+   * @param snapId - The snap id requesting the interface data.
+   * @param id - The interface id.
+   * @returns The interface state.
+   */
+  getInterface(snapId: string, id: string) {
+    this.#validateArgs(snapId, id);
+
+    return this.state.interfaces[id];
+  }
+
+  /**
+   * Updates the interface with the given content.
+   *
+   * @param snapId - The snap id requesting the update.
+   * @param id - The interface id.
+   * @param content - The new content.
+   * @returns Null.
+   */
+  updateInterface(snapId: string, id: string, content: Component) {
+    this.#validateArgs(snapId, id);
+
+    const oldState = this.state.interfaces[id].state;
+
+    const newState = constructState(oldState, content);
+
+    this.update((draftState) => {
+      draftState.interfaces[id].state = newState;
+      draftState.interfaces[id].content = content;
+    });
+
+    return null;
```

Comment:
```suggestion
```

Couldn't this just be a void function? We can return null in the RPC implementation if need be, it doesn't need to leak into the controller imo